test/e2e/test_connection.py

```diff
@@ -51,15 +57,16 @@ def test_post(self):
             }
         }
         """
-        con = Connection('http://0.0.0.0:3333')
-        con.login('root@example.com', 'test')
-        res = con.post('/v2/resources', resinfo)
+
+        res = self.con.post('/v2/resources', res_info)
+
         self.assertIsNotNone(res['@id'])
-        id = res['@id']
         self.assertEqual(res['@type'], 'anything:Thing')
         self.assertEqual(res['rdfs:label'], 'knora-py thing')
 
-        eraseinfo = """{
+        res_id = res['@id']
+
+        erase_info = """{
             "@id" : "%s",
```

Comment:
personally, I have a huge dislike for these old school C-style formatted strings.  I much prefer `f"...{id}"`

Reply:
replaced it ;)